tools/codegen/gen.py

```diff
@@ -456,9 +456,97 @@ def compute_meta_function_declaration(g: NativeFunctionsGroup) -> Optional[str]:
         parent_class = g.out.structured_inherits
         if parent_class is None:
             parent_class = "at::impl::MetaBase"
+        meta_return = "void"
+        precomputed = g.out.precomputed if g.structured else None
+
+        if precomputed:
```

Comment:
The code added below looks really ugly:
```
struct TORCH_API structured_avg_pool2d : public at::impl::MetaBase {
  
                template <bool A = false, bool B = false, bool C = false, bool D
                struct TORCH_API precompute_out {
                    
                    precompute_out<true, B, C, D, E, F> set_kH(int64_t value) {
                        TORCH_INTERNAL_ASSERT(A == false);
                        precompute_out<true, B, C, D, E, F> ret;
ret.kH = value;     
ret.kW = this->kW;      
ret.dH = this->dH;      
ret.dW = this->dW;
ret.padH = this->padH;
ret.padW = this->padW;  
return ret;
                    }
```

Is there a way to indent these blocks properly (other than adding tab characters manually)?